src/Aspire.Hosting/Extensions/ResourceBuilderExtensions.cs

```diff
@@ -328,6 +328,38 @@ private static void ApplyEndpoints<T>(this IResourceBuilder<T> builder, IResourc
         return builder.WithAnnotation(annotation);
     }
 
+    /// <summary>
+    /// Exposes an HTTP endpoint on a resource. This binding reference can be retrieved using <see cref="GetEndpoint{T}(IResourceBuilder{T}, string)"/>.
+    /// The binding name will be "http" if not specified.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="hostPort">The host port.</param>
+    /// <param name="name">The name of the endpoint.</param>
+    /// <param name="env">The name of the environment variable to inject.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <exception cref="DistributedApplicationException">Throws an exception if the a binding with the same name already exists on the specified resource.</exception>
+    public static IResourceBuilder<T> WithHttpEndpoint<T>(this IResourceBuilder<T> builder, int? hostPort = null, string? name = null, string? env = null) where T : IResource
```

Comment:
Would it make sense for `hostPort` to default to `80`?

Reply:
We can't have a default host post, it'll conflict too easily. null means use a random port